src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +287,61 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            var errorMsg = "Failed to create ICompletionExtensionProvider";
+            
+            try
+            {
+                var serviceProvider = ExtensionServiceProvider.AddAssembly(param.Assembly);
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationToken = new CancellationTokenSource(5000).Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken);
+                    if (ext == null)
+                    {
+                        errorMsg = "Failed to create ICompletionExtension";
+                        await requestContext.SendError(errorMsg);
+                        return;
+                    }
+
+                    string extName = null;
+                    try
+                    {
+                        extName = ext.Name;
+                        await ext.Initialize(cancellationToken);
+                    }
+                    catch (NotImplementedException)
+                    {
+                    }
+                    catch (NotSupportedException)
+                    {
+                    }
```

Comment:
Are these expected or something? Seems like you shouldn't be silently ignoring these

Reply:
The intention was the Initialize method doesn't have to be implemented.

Reply:
Is this standard practice for you? I would generally think that if someone didn't need it they would just have it be a no-op - exceptions should be reserved for actual errors.

Reply:
+1. It would be very counter-intuitive to have the NotImplemented and NotSupported exception caught and eaten up like it. It is better to make the Initialize method be required to be implemented which could be an no-op implementation.

Reply:
This is debatable, but I'll remove the try/catch here.